core/performance_test/peakflops/03_Cabana_peakflops.cpp

```diff
@@ -105,49 +100,72 @@ void movePx(data_t *__restrict__ a,  data_t *__restrict__ x0,
+        ParticleList& x3, ParticleList& x4, ParticleList& x5, ParticleList& x6,
+        ParticleList& x7, ParticleList& x8, ParticleList& x9, ParticleList& c,
+        long n, int num_struct )
 {
     long i;
     int s;
     int j;
 
     asm volatile ("# ax+c loop begin");
-    for(i = 0; i<n; i++)
-    {
         for ( s = 0; s < num_struct; ++s )
         {
-            for(j=s*VECLENTH; j<(s+1)*VECLENTH; j++)
+            for(i = 0; i<n; i++)
             {
-                x0(j) = a(j)*x0(j)+ c(j);
-                x1(j) = a(j)*x1(j)+ c(j);
-                x2(j) = a(j)*x2(j)+ c(j);
-                x3(j) = a(j)*x3(j)+ c(j);
-                x4(j) = a(j)*x4(j)+ c(j);
-                x5(j) = a(j)*x5(j)+ c(j);
-                x6(j) = a(j)*x6(j)+ c(j);
-                x7(j) = a(j)*x7(j)+ c(j);
-                x8(j) = a(j)*x8(j)+ c(j);
-                x9(j) = a(j)*x9(j)+ c(j);
+#pragma omp simd
+                for(j=0; j<VECLENTH; j++)
+                {
+                    auto _c = c.access(s).get<0>(j);
+                    auto _a = a.access(s).get<0>(j);
+
+                    x0.access(s).get<0>(j) = _a * x0.access(s).get<0>(j) + _c;
+                    x1.access(s).get<0>(j) = _a * x1.access(s).get<0>(j) + _c;
+                    x2.access(s).get<0>(j) = _a * x2.access(s).get<0>(j) + _c;
+                    x3.access(s).get<0>(j) = _a * x3.access(s).get<0>(j) + _c;
+                    x4.access(s).get<0>(j) = _a * x4.access(s).get<0>(j) + _c;
+                    x5.access(s).get<0>(j) = _a * x5.access(s).get<0>(j) + _c;
+                    x6.access(s).get<0>(j) = _a * x6.access(s).get<0>(j) + _c;
+                    x7.access(s).get<0>(j) = _a * x7.access(s).get<0>(j) + _c;
+                    x8.access(s).get<0>(j) = _a * x8.access(s).get<0>(j) + _c;
+                    x9.access(s).get<0>(j) = _a * x9.access(s).get<0>(j) + _c;
+                }
             }
-        }
     }
     asm volatile ("# ax+c loop end");
+    /*
     for ( s = 0; s < num_struct; ++s )
     {
-        for(j=s*VECLENTH; j<(s+1)*VECLENTH; j++)
+        for(j=0; j<VECLENTH; j++)
         {
-            x0(j) = x0(j)+x1(j)+x2(j)+x3(j)+x4(j)+x5(j)+x6(j)+x7(j)+x8(j)+x9(j);
+            x0[s].vec[j] = x0[s].vec[j]+x1[s].vec[j]+x2[s].vec[j]+x3[s].vec[j]+x4[s].vec[j]+
+                           x5[s].vec[j]+x6[s].vec[j]+x7[s].vec[j]+x8[s].vec[j]+x9[s].vec[j];
         }
     }
+    */
 }
 
-//---------------------------------------------------------------------------//
+
+
+#pragma omp declare simd simdlen(VECLENTH) uniform(s) linear(i) uniform(a)
+template<typename SliceType>
+float access_vector_mul_direct(SliceType& x, SliceType& a, SliceType& c, int s, int i)
+{
+    return a(i) * x(i) + c(i);
+}
+
+
+#pragma omp declare simd simdlen(VECLENTH) uniform(s) linear(i) uniform(a) uniform(c) uniform(x)
+template<typename SliceType>
+float access_vector_mul(SliceType& x, SliceType& a, SliceType& c, int s, int i)
```

Comment:
This function does not seem to get used.

Reply:
removed